x-pack/legacy/plugins/infra/public/components/formatted_time.tsx

```diff
@@ -17,13 +17,33 @@ const getFormattedTime = (
   return userFormat ? moment(time).format(userFormat) : moment(time).format(fallbackFormat);
 };
 
-export const useFormattedTime = (time: number, fallbackFormat?: string) => {
-  const [dateFormat] = useKibanaUiSetting('dateFormat');
-  const formattedTime = useMemo(() => getFormattedTime(time, dateFormat, fallbackFormat), [
+interface UseFormattedTimeOptions {
+  show?: 'date' | 'time' | 'all';
+  fallbackFormat?: string;
+}
+
+export const useFormattedTime = (time: number, options: UseFormattedTimeOptions = {}) => {
+  const opts: UseFormattedTimeOptions = {
+    show: 'all',
+    ...options,
+  };
+
+  let [dateFormat] = useKibanaUiSetting('dateFormat');
+
+  switch (opts.show) {
+    case 'time':
+      dateFormat = 'HH:mm:ss.SSS';
+      break;
+    case 'date':
+      dateFormat = 'MMM D, YYYY';
+      break;
+  }
```

Comment:
How about replacing this with a map of type to format?

```typescript
const formatMap: Record<'date' | 'time' | 'all', string> = {
  time: 'HH:mm:ss.SSS',
  date: 'MMM D, YYYY',
  all: useKibanaUiSetting('dateFormat')[0]
}

const dateFormat = formatMap[opts.show]
```

There's lots options for how to write/type it, but I think a key/value pairing makes it easier to see the intent

Reply:
Makes sense :+1: